fission/src/systems/scene/SceneRenderer.ts

```diff
@@ -151,6 +155,10 @@ class SceneRenderer extends WorldSystem {
             )
         })
 
+        // Update the tags each frame if they are enabled in preferences
+        if (PreferencesSystem.getGlobalPreference<boolean>("RenderSceneTags"))
```

Comment:
So, in my mind, this shouldn't be in SceneRenderer. You should have a useEffect inside the SceneOverlay component that adds an event listener to the PreferenceSystem, and will update isDisabled accordingly.